starter-code/src/viking.js

```diff
@@ -1,11 +1,98 @@
 // Soldier
-function Soldier () {}
+function Soldier (health, strength) {
+    this.health = health;
+    this.strength = strength;
+
+    this.attack = function(){
+        return this.strength;
+    };
+
+    this.receiveDamage = function(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-function Viking () {}
+function Viking (name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+
+    this.receiveDamage = function(damage) {
+        this.health -= damage;
+        if(health > 0){
+            return this.name + " has received " + this.damage + " points of damage";
+        }
+        else {
+            return this.name + " has died in act of combat";
+        }
+    }
+
+    this.battleCry = function(){
+        return "Odin Owns You All!";
+    }
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
 
 // Saxon
-function Saxon () {}
+function Saxon (health, strength) {
+    Soldier.call(this, health, strength);
+
+    this.receiveDamage = function(damage) {
+        this.health -= damage;
+        if(health > 0){
+            return "A Saxon has received " + this.damage + " points of damage";
+        }
+        else {
+            return "A Saxon has died in act of combat";
+        }
+    }
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
 
 // War
-function War () {}
+function War () {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+
+    this.addViking = function(viking) {
+        this.vikingArmy.push(viking);
+    }
+
+    this.saxonArmy = function(saxon){
+        this.saxonArmy.push(saxon);
+    }
+
+    this.vikingAttack = function(){
+        var theVictm = Math.floor(Math.random() * saxonArmy.length); 
+        var isDead = saxonArmy[theVictm].receiveDamage(300);
+        if (isDead = "A Saxon has died in act of combat"){
+            this.saxonArmy.splice(theVictm,1);
+        }
+
+        return isDead;
+    }
+
+    this.saxonAttack = function(){
```

Comment:
This code looks exactly the same as vikingAttack(). Maybe you can use a function...